<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Collection Of Human And Mouse TF Regulons • dorothea</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Collection Of Human And Mouse TF Regulons">
<meta property="og:description" content="This package contains human and mouse TF regulons.
    The human regulons were curated and collected from different types of 
    evidence such as literature curated resources, ChIP-seq peaks, TF binding 
    site motifs and interactions inferred directly from gene expression. The 
    mouse regulons were constructed by mapping the human gene symbols to their 
    orthologs in mice. Those regulons can be coupled with any statistical method
    that aims to analyse gene sets to infer TF activity from gene expression 
    data. Preferably the statistical method viper is used.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-7"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-7');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">dorothea</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/bulk_vignette.html">TF activity inference from bulk transcriptomic data with DoRothEA as regulon resource.</a>
    </li>
    <li>
      <a href="articles/single_cell_vignette.html">TF activity inference from scRNA-seq data with DoRothEA as regulon resource.</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/dorothea/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="dorothea-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#dorothea-" class="anchor"></a>dorothea <img src="reference/figures/tool_logo.png" align="right" height="139">
</h1></div>
<!-- badges: start -->

<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>DoRothEA is a gene set resource containing signed transcription factor (TF) - target interactions first described in <a href="https://doi.org/10.1101/gr.240663.118">Garcia-Alonso et al., 2019</a>. The collection of a TF and its transcriptional targets is defined as regulon. DoRothEA regulons were curated and collected from different types of evidence such as literature curated resources, ChIP-seq peaks, TF binding site motifs and interactions inferred directly from gene expression.</p>
<p><img src="reference/figures/overview.png" align="center" width="600"></p>
<p>For each TF-target interaction we assigned a confidence level based on the number of supporting evidence. The confidence assignment comprises five levels, ranging from A (highest confidence) to E (lowest confidence). Interactions that are supported by all four lines of evidence, manually curated by experts in specific reviews, or supported both in at least two curated resources are considered to be highly reliable and were assigned an A level. Level B-D are reserved for curated and/or ChIP-seq interactions with different levels of additional evidence. Finally, E level is used for interactions that are uniquely supported by computational predictions. To provide the most confident regulon for each TF, we aggregated the TF-target interactions with the highest possible confidence score that resulted in a regulon size equal to or greater than ten targets. The final confidence level assigned to the TF regulon is the lowest confidence score of its component targets.</p>
<p>DoRothEA regulons can be coupled with several statistical method yielding a <em>functional analysis</em>-tool to infer TF activity from gene expression data. The activity is computed by considering not the gene expression of the TFs itself but the mRNA levels of their direct transcriptional targets. We define the transcriptional targets to as footprints of a TF on gene expression. A more detailed description of the concept of <em>footprint-based</em> analysis is available in the review <a href="https://doi.org/10.1016/j.coisb.2019.04.002">Dugourd et al., 2019</a>.</p>
<p>Typically, DoRothEA is coupled with the statistical method <a href="https://www.bioconductor.org/packages/release/bioc/html/viper.html">VIPER</a> as it incorporates the mode of regulation of each TF-target interaction. However, VIPER can be replaced by any other statistical method that aims to analyze gene sets, e.g. GSEA.</p>
<p>We provide for human and mouse, respectively, two different DoRothEA version’s that differ solely in the analyzed gene expression data for the network inference evidence. For the standard versions <code>dorothea_hs</code>/<code>dorothea_mm</code>, we analyzed tissue specific gene expression data from the <a href="https://www.gtexportal.org/home/">GTEx portal</a>. For the versions <code>dorothea_hs_pancancer</code>/<code>dorothea_mm_pancancer</code> we used gene expression data of several primary cancers from the <a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga">TCGA programm</a> which is supposed to make DoRothEA more applicable to cancer-related studies.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install from bioconductor</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"dorothea"</span><span class="op">)</span>

<span class="co"># install the development version from GitHub</span>
<span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"saezlab/dorothea"</span><span class="op">)</span></code></pre></div>
</div>
<div id="more-information" class="section level2">
<h2 class="hasAnchor">
<a href="#more-information" class="anchor"></a>More information</h2>
<p>DoRothEA is also part of our <a href="https://github.com/saezlab/transcriptutorial">transcripttutorial</a>. There you will find additional information on how DoRothEA’s regulons can be used to analyze RNA-seq data and how TF activities can help to contextualize signaling networks.</p>
</div>
<div id="updates" class="section level2">
<h2 class="hasAnchor">
<a href="#updates" class="anchor"></a>Updates</h2>
<div id="update-1---extension-to-mouse" class="section level3">
<h3 class="hasAnchor">
<a href="#update-1---extension-to-mouse" class="anchor"></a>Update #1 - extension to mouse</h3>
<p>Originally DoRothEA contained only human regulons. In a benchmark study we showed that DoRothEA in combination with VIPER is also applicable to mouse data, as described in <a href="https://doi.org/10.1016/j.bbagrm.2019.194431">Holland et al., 2019</a>. Accordingly, we developed a mouse version of DoRothEA by transforming the human genes to their mouse orthologs.</p>
</div>
<div id="update-2---extension-to-single-cell-rna-seq-data" class="section level3">
<h3 class="hasAnchor">
<a href="#update-2---extension-to-single-cell-rna-seq-data" class="anchor"></a>Update #2 - extension to single-cell RNA-seq data</h3>
<p>Recent technological advances in single-cell RNA-seq enable the profiling of gene expression at the individual cell level. We showed that DoRothEA in combination with VIPER can be applied to scRNA-seq data, as described in <a href="https://doi.org/10.1186/s13059-020-1949-z">Holland et al., 2020</a>.</p>
</div>
</div>
<div id="citing-dorothea" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-dorothea" class="anchor"></a>Citing DoRothEA</h2>
<p>Beside the original paper there are two additional papers expanding the usage of DoRothEA regulons.</p>
<ul>
<li>
<p>If you use DoRothEA for your research please cite the original publication:</p>
<blockquote>
<p>Garcia-Alonso L, Holland CH, Ibrahim MM, Turei D, Saez-Rodriguez J. “Benchmark and integration of resources for the estimation of human transcription factor activities.” <em>Genome Research.</em> 2019. DOI: <a href="https://doi.org/10.1101/gr.240663.118">10.1101/gr.240663.118</a>.</p>
</blockquote>
</li>
<li>
<p>If you use the mouse version of DoRothEA please cite additionally:</p>
<blockquote>
<p>Holland CH, Szalai B, Saez-Rodriguez J. “Transfer of regulatory knowledge from human to mouse for functional genomics analysis.” <em>Biochimica et Biophysica Acta (BBA) - Gene Regulatory Mechanisms.</em> 2019. DOI: <a href="https://doi.org/10.1016/j.bbagrm.2019.194431">10.1016/j.bbagrm.2019.194431</a>.</p>
</blockquote>
</li>
<li>
<p>If you apply DoRothEA’s regulons on single-cell RNA-seq data please cite additionally:</p>
<blockquote>
<p>Holland CH, Tanevski J, Perales-Patón J, Gleixner J, Kumar MP, Mereu E, Joughin BA, Stegle O, Lauffenburger DA, Heyn H, Szalai B, Saez-Rodriguez, J. “Robustness and applicability of transcription factor and pathway analysis tools on single-cell RNA-seq data.” <em>Genome Biology.</em> 2020. DOI: <a href="https://doi.org/10.1186/s13059-020-1949-z">10.1186/s13059-020-1949-z</a>.</p>
</blockquote>
</li>
</ul>
</div>
<div id="usage-of-dorothea-for-commercial-purpose" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-of-dorothea-for-commercial-purpose" class="anchor"></a>Usage of DoRothEA for commercial purpose</h2>
<p>DoRothEA as it stands is intended only for academic use as in contains resources whose licenses don’t permit commerical use. However, we developed a non-academic version of DoRothEA by removing those resources (mainly TRED from the curated databases). You find the non-academic package with the regulons <a href="https://github.com/saezlab/dorothea/tree/non-academic">here</a>.</p>
</div>
<div id="history" class="section level2">
<h2 class="hasAnchor">
<a href="#history" class="anchor"></a>History</h2>
<p>Initially DoRothEA was a framework of <strong>D</strong>iscriminant <strong>R</strong>egulon <strong>E</strong>xpression <strong>A</strong>nalysis, as described in <a href="https://cancerres.aacrjournals.org/content/early/2017/12/09/0008-5472.CAN-17-1679">Garcia-Alonso et al., 2017</a>. At that time DoRothEA was used with a smaller set of regulons to find associations between TF activities and drug responses. Those results are queryable at <a href="https://dorothea.opentargets.io/">Open Targets</a>. Over the past years DoRothEA evolved from the analytical framework to a pure regulon resource as it is now.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/dorothea">https://​www.bioconductor.org/​packages/​dorothea</a>
</li>
<li>Browse source code at <br><a href="https://github.com/saezlab/dorothea/">https://​github.com/​saezlab/​dorothea/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/saezlab/dorothea/issues">https://​github.com/​saezlab/​dorothea/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing dorothea</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Daniel Dimitrov <br><small class="roles"> Maintainer </small> <a href="https://orcid.org/0000-0002-5197-2112" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Christian H. Holland <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-3060-5786" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Luz Garcia-Alonso <br><small class="roles"> Author </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/saezlab/dorothea/actions"><img src="https://github.com/saezlab/dorothea/workflows/R-CMD-check-bioc/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://bioconductor.org/checkResults/release/data-experiment-LATEST/dorothea"><img src="http://bioconductor.org/shields/build/release/data-experiment/dorothea.svg" alt="BioC status"></a></li>
<li><a href="https://codecov.io/gh/saezlab/dorothea"><img src="https://codecov.io/gh/saezlab/dorothea/branch/master/graph/badge.svg" alt="codecov"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Daniel Dimitrov, Christian H. Holland, Luz Garcia-Alonso.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
